CMakeLists.txt

```diff
@@ -118,6 +118,12 @@ if(${GFXRECON_ENABLE_RELEASE_ASSERTS})
     add_definitions(-DGFXRECON_ENABLE_RELEASE_ASSERTS)
 endif()
 
+option(GFXRECON_ENABLE_ASAN "Use address sanitization")
+if (GFXRECON_ENABLE_ASAN)
+    add_compile_options(-fsanitize=address -fno-omit-frame-pointer)
+    add_link_options(-fsanitize=address)
```

Comment:
It seems like you'd want to have different versions of this for MS and other platforms like this example shows:
https://cmake.org/cmake/help/latest/command/add_compile_options.html#example

Something like this using the forward slash option notation for windows:
```
if (MSVC)
    add_compile_options(/fsanitize=address /Oy-)
else()
# as before ...
endif()
```
I'm the last person to speak confidently on build system stuff though and I probably missed something (does cmake translate the options to what MSVC expects or is it clang being fed these options even on windows by some other mechanism perhaps...).

Reply:
Although what we might reall want is the CMake presets from CMake 3.19:
https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html
E.g.:
https://github.com/cpp-best-practices/cmake_template/blob/main/CMakePresets.json